src/admin/mod.rs

```diff
@@ -185,6 +197,18 @@ async fn handle_pprof(_req: Request<Body>) -> Response<Body> {
     }
 }
 
+async fn handle_server_shutdown(
```

Comment:
I think for safety we will want to make this POST only

Reply:
sure, that makes sense!